### PR TITLE
Build the plugin in Sonatype CI - CLM-28108

### PR DESCRIPTION
**Pull request by eduard-tita**
#### Links
Jira: https://sonatype.atlassian.net/browse/CLM-28108
Build: https://jenkins.ci.sonatype.dev/job/integrations/job/jenkins/job/feature-snapshots/job/CLM-28108-build-it-in-Sonatype-CI/

### CODE REVIEW

**Review by koraytugay — APPROVED** [review]
+1 not very confident but lgtm

File: Jenkinsfile.sonatype [hunk 1]
```diff
@@ -4,63 +4,37 @@
...
+            'target/nexus-jenkins-plugin.pom'
+        ]
+    ],
+    performSonarAnalysis             : true,
+    sonarAnalysisPullRequestsOnly    : false,
+    downstreamJobName                : 'extra-tests',
+    prepare                          : { build, env ->
+      githubStatusUpdate('pending')
+    },
+    releaseRetentionPolicy           : RetentionPolicy.TEN_BUILDS,
```
Comment (juanmafabbri) [inline Jenkinsfile.sonatype:36]: Does this mean that there will be up to 10 versions? If there are more, will they be deleted?

File: pom.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
   <groupId>org.sonatype.nexus.ci</groupId>
   <artifactId>nexus-jenkins-plugin</artifactId>
-  <version>${revision}.${changelist}</version>
+  <version>3.18.0-SNAPSHOT</version>
```
Comment (juanmafabbri) [inline pom.xml:28]: question, where is it defined that the release number is taken from this value?

File: Jenkinsfile.sonatype (hunk 1 above)
Comment (eduard-tita) [reply Jenkinsfile.sonatype:36]: Yes. We will retain the latest 10 builds

File: pom.xml (hunk 2 above)
Comment (eduard-tita) [reply pom.xml:28]: The release version is user entered - see the new release job parameters: https://jenkins.ci.sonatype.dev/job/integrations/job/jenkins/job/release/

File: Jenkinsfile.sonatype (hunk 1 above)
Comment (vladt) [inline Jenkinsfile.sonatype:36]: Usually, this is specified in Jenkins, not in the jenkinsfile

File: pom.xml (hunk 2 above)
Comment (juanmafabbri) [reply pom.xml:28]: from here, right?
image

File: Jenkinsfile.sonatype (hunk 1 above)
Comment (juanmafabbri) [reply Jenkinsfile.sonatype:36]: is it related to this? 
```
buildDiscarder(
        logRotator(numToKeepStr: '100', artifactNumToKeepStr: '20')
)
```

Comment (eduard-tita) [reply Jenkinsfile.sonatype:36]: True and the above gets overwritten if anything is specified in Jenkins. 

The above line is present in all integrations pipelines and provided a better default value.

Comment (vladt) [reply Jenkinsfile.sonatype:36]: > True and the above gets overwritten if anything is specified in Jenkins.

Are you sure what's specified in Jenkins has priority?

I remember I had to remove such settings from jenkinsfile to make them work in Jenkins. Maybe I'm misremembering...

Comment (eduard-tita) [reply Jenkinsfile.sonatype:36]: > Are you sure what's specified in Jenkins has priority?

I think so, but I'm not sure. Regardless, 10 builds is what we want to set for release jobs anyways.

**Review by juanmafabbri — APPROVED** [review]
+1

File: Jenkinsfile.sonatype (hunk 1 above)
Comment (tneer) [reply Jenkinsfile.sonatype:36]: @vladt @eduard-tita @juanmafabbri  Retention settings are always controlled by the Jenkinsfile when using our standard Maven pipelines (you can change them in the UI, but they will get overwritten the next time the job runs).  The default for `releaseRetentionPolicy` is 100 builds for a max of 14 days (https://github.com/sonatype/jenkins-shared/blob/35599a6324d5871e08735892281a4362ed5f836f/vars/RetentionPolicy.groovy#L18).   After 14 days, only 1 build will be retained. Using the TEN_BUILDS policy will keep up to 10 builds indefinitely, which is usually what is desired for release builds.  Occasionally folks keep 100 release builds.

**Pull request merged**